src/main/java/com/sap/oss/phosphor/fosstars/data/github/GithubInnerSourceTopicDataProvider.java

```diff
@@ -0,0 +1,54 @@
+package com.sap.oss.phosphor.fosstars.data.github;
+
+import static com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures.HAS_INNERSOURCE_TOPIC;
+
+import com.sap.oss.phosphor.fosstars.model.Feature;
+import com.sap.oss.phosphor.fosstars.model.Value;
+import com.sap.oss.phosphor.fosstars.model.ValueSet;
+import com.sap.oss.phosphor.fosstars.model.other.Utils;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.GitHubProject;
+import com.sap.oss.phosphor.fosstars.model.value.ValueHashSet;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+
+public class GithubInnerSourceTopicDataProvider extends GitHubCachingDataProvider {
+
+  private static final Set<String> ACCEPTED_TOPICS;
+
+  static {
+    Set<String> acceptedTopics = new HashSet<>();
+    acceptedTopics.add("inner-source");
+    acceptedTopics.add("innersource");
+    ACCEPTED_TOPICS = Collections.unmodifiableSet(acceptedTopics);
+  }
+
+  public GithubInnerSourceTopicDataProvider(GitHubDataFetcher fetcher) {
+    super(fetcher);
+  }
+
+  @Override
+  public Set<Feature<?>> supportedFeatures() {
+    return Utils.setOf(HAS_INNERSOURCE_TOPIC);
+  }
+
+  @Override
+  protected ValueSet fetchValuesFor(GitHubProject project) throws IOException {
+    Optional<String> topicOptional = fetcher.githubTopicsFor(project).stream().filter(topic -> {
+      return ACCEPTED_TOPICS.contains(topic.toLowerCase(Locale.US));
+    }).findFirst();
+
+    Value<Boolean> value;
+    if (topicOptional.isPresent()) {
+      value = HAS_INNERSOURCE_TOPIC.value(true);
+    } else {
+      value = HAS_INNERSOURCE_TOPIC.value(false)
+        .explain("The repository does not have an InnerSource topic.");
+    }
+
+    return ValueHashSet.from(value);
+  }
+}
```

Comment:
```suggestion
package com.sap.oss.phosphor.fosstars.data.github;

import static com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures.HAS_INNERSOURCE_TOPIC;

import com.sap.oss.phosphor.fosstars.model.Feature;
import com.sap.oss.phosphor.fosstars.model.Value;
import com.sap.oss.phosphor.fosstars.model.subject.oss.GitHubProject;
import java.io.IOException;
import java.util.Arrays;
import java.util.HashSet;
import java.util.Locale;
import java.util.Set;

public class HasInnerSourceTopic extends CachedSingleFeatureGitHubDataProvider<Boolean> {

  private static final Set<String> ACCEPTED_TOPICS = new HashSet<>(Arrays.asList("inner-source", "innersource"));

  public HasInnerSourceTopic(GitHubDataFetcher fetcher) {
    super(fetcher);
  }

  @Override
  public Feature<Boolean> supportedFeature() {
    return HAS_INNERSOURCE_TOPIC;
  }

  @Override
  protected Value<Boolean> fetchValueFor(GitHubProject project) throws IOException {
    return HAS_INNERSOURCE_TOPIC.value(fetcher.githubTopicsFor(project).stream()
    .anyMatch(HasInnerSourceTopic::found))
    .explainIf(false, "The repository does not have an InnerSource topic.");
  }

  private static boolean found(String topic) {
    return ACCEPTED_TOPICS.contains(topic.toLowerCase(Locale.US));
  }
}
```
I have provided a gist of what we can use if only a single feature is present and this feature is extracted from GitHub

Also please add Java Docs and Logs where necessary maybe keep `HasSecurityTeam.java`